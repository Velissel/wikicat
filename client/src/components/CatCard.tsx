import React from 'react';
import { Button, Card, CardImg } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { CatData } from 'src/data/searchCatsByBreedName';
// @ts-ignore
import styles from './CatCard.module.scss';

interface CatCardProps {
  cat: CatData;
}
export default function CatCard(props: CatCardProps) {
  const { id, image, name, description } = props.cat;
  return (
    <Card>
      <CardImg
        variant="top"
        src={image}
        className={styles.image}
        data-testid="cat-img"
      ></CardImg>
      <Card.Body>
        <Card.Title>{name}</Card.Title>
        <Card.Text className={styles.description}>{description}</Card.Text>
        <Link to={`/kitty/${id}`}>
          <Button>View Breed Details</Button>
        </Link>
      </Card.Body>
    </Card>
  );
}
