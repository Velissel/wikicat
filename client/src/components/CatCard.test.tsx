import React from 'react';
import { render, screen } from '@testing-library/react';
import CatCard from './CatCard';
import { MemoryRouter } from 'react-router-dom';

describe('CatCard component', () => {
  it('should render', async () => {
    const cat = {
      id: 'id',
      image: 'url',
      name: 'name',
      description: 'description',
    };
    const { container } = render(
      <MemoryRouter>
        <CatCard cat={cat} />
      </MemoryRouter>,
    );
    expect(screen.getByText(cat.name)).toBeInTheDocument();
    expect(screen.getByText(cat.description)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: 'View Breed Details' }),
    ).toHaveAttribute('href', `/kitty/${cat.id}`);
    expect(screen.getByTestId('cat-img')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
