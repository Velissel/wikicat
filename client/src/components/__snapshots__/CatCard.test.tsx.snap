// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCard component should render 1`] = `
<div>
  <div
    class="card"
    data-testid="cat-card-id"
  >
    <img
      class="card-img-top image"
      data-testid="cat-img"
      src="url"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        name
      </div>
      <p
        class="description card-text"
      >
        description
      </p>
      <a
        href="/kitty/id"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          View Breed Details
        </button>
      </a>
    </div>
  </div>
</div>
`;
